radiospectra/net/sources/psp.py

```diff
@@ -0,0 +1,137 @@
+            if wavelength.min in RECEIVER_FREQUENCIES['rfs_hfr'] or wavelength.max in RECEIVER_FREQUENCIES['rfs_hfr']:
+                receivers.append('rfs_hfr')
+            if wavelength.min in RECEIVER_FREQUENCIES['rfs_lfr'] or wavelength.max in RECEIVER_FREQUENCIES['rfs_lfr']:
+                receivers.append('rfs_lfr')
+            # min in lfr and max in hfr
+            # min and max of combined lft and hfr contained in give wavelength range
+            if a.Wavelength(RECEIVER_FREQUENCIES['rfs_lfr'].min,
+                            RECEIVER_FREQUENCIES['rfs_hfr'].max) in wavelength:
+                receivers = ['rfs_lfr', 'rfs_hfr']
+            # If we get here the is no overlap so set to empty list
+        return receivers
+
+    def search(self, *args, **kwargs):
+        """
+         Query this client for a list of results.
+
+         Parameters
+         ----------
+         *args: `tuple`
+             `sunpy.net.attrs` objects representing the query.
+         **kwargs: `dict`
+              Any extra keywords to refine the search.
+
+         Returns
+         -------
+         A `QueryResponse` instance containing the query result.
+         """
+        matchdict = self._get_match_dict(*args, **kwargs)
+        req_wave = matchdict.get('Wavelength', None)
+        receivers = RECEIVER_FREQUENCIES.keys()
+        if req_wave is not None:
+            receivers = self._check_wavelengths(req_wave)
+
+        metalist = []
+        start_year = matchdict['Time'].start.datetime.year
+        end_year = matchdict['Time'].end.datetime.year
+        for receiver in receivers:
+            for year in range(start_year, end_year+1):
+                urlpattern = self.baseurl.format(Wavelength=receiver, year=year)
+                scraper = Scraper(urlpattern, regex=True)
+                filesmeta = scraper._extract_files_meta(matchdict['Time'], extractor=self.pattern)
+                for i in filesmeta:
+                    rowdict = self.post_search_hook(i, matchdict)
+                    metalist.append(rowdict)
+
+        return QueryResponse(metalist, client=self)
+
+    def post_search_hook(self, exdict, matchdict):
+        """
+        This method converts 'rfs_hfr' and 'rfs_lfr' in the url's metadata
+        to the frequency ranges of for low and high frequency receivers.
+        """
+        rowdict = super().post_search_hook(exdict, matchdict)
+        if rowdict['Wavelength'] == 'rfs_hfr':
+            fr = RECEIVER_FREQUENCIES['rfs_hfr']
+            rowdict['Wavelength'] = u.Quantity([float(fr.min.value), float(fr.max.value)], unit=fr.unit)
+        elif rowdict['Wavelength'] == 'rfs_lfr':
+            fr = RECEIVER_FREQUENCIES['rfs_lfr']
+            rowdict['Wavelength'] = u.Quantity([float(fr.min.value), float(fr.max.value)], unit=fr.unit)
+        return rowdict
+
+    @classmethod
+    def register_values(cls):
+        from sunpy.net import attrs
```

Comment:
this is already imported above

Reply:
Yea it's like that in all the sunpy clients so I thought it was necessary, I'll remove and see what happens

Reply:
It was needed in sunpy core since we have a circular import otherwise.

Reply:
Removed